core/shared/src/main/scala/magnolia.scala

```diff
@@ -238,34 +255,39 @@ object Magnolia {
 
         val preAssignments = caseParams.map(_.typeclass)
 
-        val defaults = if (!isValueClass) {
-          val companionRef = GlobalUtil.patchedCompanionRef(c)(genericType.dealias)
-          val companionSym = companionRef.symbol.asModule.info
-
-          // If a companion object is defined with alternative apply methods
-          // it is needed get all the alternatives
-          val constructorMethods =
-            companionSym.decl(TermName("apply")).alternatives.map(_.asMethod)
 
-          // The last apply method in the alternatives is the one that belongs
-          // to the case class, not the user defined companion object
-          val indexedConstructorParams =
-            constructorMethods.last.paramLists.head.map(_.asTerm).zipWithIndex
+        val defaults = headParamList.filterNot(_ => isValueClass) map { plist =>
```

Comment:
The `filterNot` on an external value looks a bit unconventional... Not hugely fussed about it, but how about just `if(isValueClass) Nil else ...`?

Reply:
Turns out that I can just lose the filter entirely, and all the tests pass just fine